modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiServiceSuite.scala

```diff
@@ -0,0 +1,149 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.effect.{IO, Resource}
+import cats.syntax.applicative._
+import cats.syntax.either._
+import cats.syntax.option._
+
+import org.tessellation.BuildInfo
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error._
+import org.tessellation.rosetta.domain.network.{BlockchainId, NetworkEnvironment}
+import org.tessellation.rosetta.domain.networkapi.NetworkApiService.LastSnapshotInfo
+import org.tessellation.rosetta.domain.networkapi.model.options
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.epoch.EpochProgress
+import org.tessellation.schema.generators._
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.timestamp.SnapshotTimestamp
+import org.tessellation.schema.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.sdk.config.AppEnvironment
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.hex.Hex
+import org.tessellation.security.signature.Signed.forAsyncKryo
+import org.tessellation.security.{KeyPairGenerator, SecurityProvider}
+import org.tessellation.shared.sharedKryoRegistrar
+
+import eu.timepit.refined.cats.{refTypeEq, refTypeOrder, refTypeShow}
+import org.scalacheck.{Arbitrary, Gen}
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object NetworkApiServiceSuite extends MutableIOSuite with Checkers {
+  type Res = (SecurityProvider[IO], KryoSerializer[IO])
+
+  override def sharedResource: Resource[IO, Res] =
+    KryoSerializer
+      .forAsync[IO](sharedKryoRegistrar)
+      .flatMap(kryo => SecurityProvider.forAsync[IO].map(securityProvider => (securityProvider, kryo)))
+
+  private def errorResult[A]: IO[A] = IO.raiseError(new Exception("unexpected call"))
+  def makeNetworkApiService(
+    lastSnapshotInfo: IO[Option[LastSnapshotInfo]] = errorResult,
+    genesisSnapshot: IO[(SnapshotOrdinal, Hash)] = errorResult,
+    nodeState: IO[NodeState] = errorResult
+  ): NetworkApiService[IO] =
+    NetworkApiService.make[IO](lastSnapshotInfo, genesisSnapshot, nodeState)
+
+  test("list returns mainnet identifier for that AppEnvironment ") { _ =>
+    val netId = NetworkIdentifier(BlockchainId.dag, NetworkEnvironment.Mainnet, none)
+    val expected = List(netId).asRight[NetworkApiError]
+    makeNetworkApiService()
+      .list(AppEnvironment.Mainnet)
+      .value
+      .map(actual => expect.eql(expected, actual))
```

Comment:
`map(expect.eql(expected, _))`

Reply:
code changed due to change in return type of `.list`